site/tests/app/controllers/admin/ConfigurationControllerTester.php

```diff
@@ -379,4 +379,72 @@ public function testUpdateConfigNonWriteable() {
             chmod($this->course_config, 0600);
         }
     }
+
+    public function testUpdateConfigurationEnableForum() {
+        $this->setUpConfig();
+        $core = new Core();
+        $controller = new ConfigurationController($core);
+        $config = new Config($core);
+        $config->loadMasterConfigs($this->master_configs_dir);
+        $config->loadCourseJson('f19', 'sample', $this->course_config);
+        $core->setConfig($config);
+        
+        $_POST['name'] = 'forum_enabled';
+        $_POST['entry'] = 'true';
+        $queries = $this->createMock(DatabaseQueries::class);
+        $queries
+            ->expects($this->once())
+            ->method('getCategories')
+            ->with()
+            ->willReturn([]);
+        $queries
+            ->expects($this->exactly(4))
+            ->method('addNewCategory')
+            ->withConsecutive(
+                [$this->equalTo('General Questions')],
+                [$this->equalTo('Homework Help')],
+                [$this->equalTo('Quizzes')],
+                [$this->equalTo('Tests')]
+            )
+            ->will($this->onConsecutiveCalls(0, 1, 2, 3));
+
+        $core->setQueries($queries);
+        $response = $controller->updateConfiguration();
+        $this->assertNotNull($response->json_response);
+    }
+
+    public function testUpdateConfigurationEnableForumWithCategories() {
+        $this->setUpConfig();
+        $core = new Core();
+        $controller = new ConfigurationController($core);
+        $config = new Config($core);
+        $config->loadMasterConfigs($this->master_configs_dir);
+        $config->loadCourseJson('f19', 'sample', $this->course_config);
+        $core->setConfig($config);
+        $_POST['name'] = 'forum_enabled';
+        $_POST['entry'] = 'true';
+        $queries = $this->createMock(DatabaseQueries::class);
+        $queries
+            ->expects($this->once())
+            ->method('getCategories')
+            ->with()
+            ->willReturn(['General Questions', 'Homework Help', 'Quizzes', 'Tests']);
```

Comment:
Let's have this return just `['Category']` to show that it does not make any calls even if none of the default categories exist.